### PR TITLE
upgrade sdk to firecracker 1.0 release

### PR DESCRIPTION
**Pull request by henry118**
Signed-off-by: Henry Wang <henwang@amazon.com>

*Issues:*
#381, #371

*Description of changes:*
Upgrade SDK to Firecracker 1.0 release

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by Kern-- — COMMENTED** [review]
Can you investigate this test failure?

```
=== RUN   TestJailerMicroVMExecution
--
  | time="2022-02-10T21:27:16Z" level=info msg="Called startVMM(), setting up a VMM on /tmp/jailer-test876952106/firecracker-main/b/root/TestJailerMicroVMExecution.socket"
  | time="2022-02-10T21:27:16Z" level=warning msg="firecracker exited: exit status 1"
  | time="2022-02-10T21:27:16Z" level=warning msg="Failed handler \"fcinit.StartVMM\": Firecracker did not create API socket /tmp/jailer-test876952106/firecracker-main/b/root/TestJailerMicroVMExecution.socket: 1 error occurred:\n\t* exit status 1\n\n"
  | machine_test.go:280: Failed to start VMM: Firecracker did not create API socket /tmp/jailer-test876952106/firecracker-main/b/root/TestJailerMicroVMExecution.socket: 1 error occurred:
  | * exit status 1
```

File: machine.go [hunk 1]
```diff
@@ -51,24 +51,19 @@ const (
...
-	SeccompLevelBasic
-	// SeccompLevelAdvanced adds further checks on some of the parameters of the
-	// allowed syscalls.
-	SeccompLevelAdvanced
-)
-
-func (level SeccompLevelValue) String() string {
-	return strconv.Itoa(int(level))
+	// Supplying user-provided, custom filters
+	CustomFilter string
```
Comment (kzys) [inline machine.go:60]: Is it a filepath to something?

File: machine.go [hunk 2]
```diff
@@ -51,24 +51,19 @@ const (
 	defaultFirecrackerInitTimeoutSeconds = 3
 )
 
-// SeccompLevelValue represents a secure computing level type.
-type SeccompLevelValue int
+// Secure computing configuration
+type SeccompConfig struct {
```
Comment (kzys) [inline machine.go:55]: Please follow https://go.dev/doc/effective_go#commentary

> The first sentence should be a one-sentence summary that starts with the name being declared.

File: machine.go [hunk 3]
```diff
@@ -51,24 +51,19 @@ const (
...
-	SeccompLevelAdvanced
-)
-
-func (level SeccompLevelValue) String() string {
-	return strconv.Itoa(int(level))
+	// Supplying user-provided, custom filters
+	CustomFilter string
 }
 
+//func (level SeccompLevelValue) String() string {
```
[the remark this answers is not in the document]
Comment (kzys) [reply machine.go:63]: +1

File: machine.go [hunk 4]
```diff
@@ -207,7 +197,7 @@ func (cfg *Config) Validate() error {
 		Int64Value(cfg.MachineCfg.MemSizeMib) < 1 {
 		return fmt.Errorf("machine needs a nonzero amount of memory")
 	}
-	if cfg.MachineCfg.HtEnabled == nil {
+	if cfg.MachineCfg.Smt == nil {
 		return fmt.Errorf("machine needs a setting for ht_enabled")
 	}
```
Comment (kzys) [inline machine.go:202]: - The error has to be updated as well
- Doesn't it have a default value?

**Comment by henry118** [comment]
> Can you investigate this test failure?
> 
> ```
> === RUN   TestJailerMicroVMExecution
> --
>   | time="2022-02-10T21:27:16Z" level=info msg="Called startVMM(), setting up a VMM on /tmp/jailer-test876952106/firecracker-main/b/root/TestJailerMicroVMExecution.socket"
>   | time="2022-02-10T21:27:16Z" level=warning msg="firecracker exited: exit status 1"
>   | time="2022-02-10T21:27:16Z" level=warning msg="Failed handler \"fcinit.StartVMM\": Firecracker did not create API socket /tmp/jailer-test876952106/firecracker-main/b/root/TestJailerMicroVMExecution.socket: 1 error occurred:\n\t* exit status 1\n\n"
>   | machine_test.go:280: Failed to start VMM: Firecracker did not create API socket /tmp/jailer-test876952106/firecracker-main/b/root/TestJailerMicroVMExecution.socket: 1 error occurred:
>   | * exit status 1
> ```

Where did you see this issue?

File: machine.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (henry118) [reply machine.go:63]: will do

File: machine.go (hunk 1 above)
Comment (henry118) [reply machine.go:60]: this is a file path

File: machine.go (hunk 2 above)
Comment (henry118) [reply machine.go:55]: will do

File: machine.go (hunk 4 above)
Comment (henry118) [reply machine.go:202]: true. i'll see if it's safe to remove this check

**Comment by henry118** [comment]
> Can you investigate this test failure?
> 
> ```
> === RUN   TestJailerMicroVMExecution
> --
>   | time="2022-02-10T21:27:16Z" level=info msg="Called startVMM(), setting up a VMM on /tmp/jailer-test876952106/firecracker-main/b/root/TestJailerMicroVMExecution.socket"
>   | time="2022-02-10T21:27:16Z" level=warning msg="firecracker exited: exit status 1"
>   | time="2022-02-10T21:27:16Z" level=warning msg="Failed handler \"fcinit.StartVMM\": Firecracker did not create API socket /tmp/jailer-test876952106/firecracker-main/b/root/TestJailerMicroVMExecution.socket: 1 error occurred:\n\t* exit status 1\n\n"
>   | machine_test.go:280: Failed to start VMM: Firecracker did not create API socket /tmp/jailer-test876952106/firecracker-main/b/root/TestJailerMicroVMExecution.socket: 1 error occurred:
>   | * exit status 1
> ```

Fixed

**Review by Kern-- — APPROVED** [review]
nit: I would prefer to squash the commits into a single commit (or maybe a base commit + as a single update commit) but I won't block on that.

File: jailer_test.go [hunk 5]
```diff
@@ -14,11 +14,21 @@ package firecracker
...
 	"context"
+	"fmt"
+	"io/ioutil"
 	"path/filepath"
 	"reflect"
+	"strings"
 	"testing"
 )
 
+func getNumaCpuset(node int) string {
```
Comment (kzys) [inline jailer_test.go:25]: How about having this function on jailer.go side? Args() is doing basically the same thing.

File: machine.go (hunk 1 above)
Comment (kzys) [reply machine.go:60]: How about calling this field `Filter`, since Firecracker side is just `--seccomp-filter`?

File: machine.go [hunk 6]
```diff
@@ -151,14 +142,9 @@ type Config struct {
...
-	// restrictive they should be. Possible values are:
-	//
-	//	0 : (default): disabled.
-	//	1 : basic filtering. This prohibits syscalls not whitelisted by Firecracker.
-	//	2 : advanced filtering. This adds further checks on some of the
-	//			parameters of the allowed syscalls.
-	SeccompLevel SeccompLevelValue
+	// SeccompCfg specifies whether seccomp filters should be installed and how
+	// restrictive they should be.
+	SeccompCfg SeccompConfig
```
Comment (kzys) [inline machine.go:147]: How about calling this field just `Seccomp`? We are sadly not really consistent about adding `Cfg` on the fields, but `config.Seccomp.Enabled = true` looks better than `config.SeccompCfg.Enabled = true`.

**Comment by kzys** [comment]
I agree with Kern that we should keep the commit history bit tidy.

File: jailer_test.go (hunk 5 above)
Comment (henry118) [reply jailer_test.go]: done.

File: machine.go (hunk 1 above)
Comment (henry118) [reply machine.go]: updated.

File: machine.go (hunk 6 above)
Comment (henry118) [reply machine.go]: updated.

**Review by kzys — APPROVED** [review]

**Pull request merged**